scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -766,6 +773,244 @@ public DatasetId storeSimulationResults(final PlanMetadata planMetadata,
     return datasetIds.datasetId();
   }
 
+  private Map<SimulatedActivityId, SimulatedActivity> getSimulatedActivities(SimulationDatasetId datasetId, Instant startSimulation)
+  throws PlanServiceException, IOException, InvalidJsonException
+  {
+    final var request = """
+        query{
+          simulated_activity(where: {simulation_dataset_id: {_eq: %d}}) {
+            activity_directive {
+              id
+              arguments
+              type
+              anchored_to_start
+              anchor_id
+            }
+            activity_type_name
+            duration
+            id
+            parent_id
+            start_offset
+            attributes
+          }
+        }
+        """.formatted(datasetId.id());
+    final JsonObject response;
+    response = postRequest(request).get();
+    final var data = response.getJsonObject("data").getJsonArray("simulated_activity");
+    return parseSimulatedActivities(data, startSimulation);
+  }
+
+private Profiles getProfiles(SimulationDatasetId datasetId) throws PlanServiceException, IOException {
+    final var request = """
+        query{
+          profile(where: {dataset_id: {_eq: %d}}){
+            type
+            profile_segments {
+              start_offset
+              dynamics
+            }
+            name
+          }
+        }
+        """.formatted(datasetId.id());
+    final JsonObject response;
+    response = postRequest(request).get();
+    final var data = response.getJsonObject("data").getJsonArray("profile");
+    return parseProfiles(data);
+}
+
+private Map<SimulatedActivityId, UnfinishedActivity> getSpans(SimulationDatasetId datasetId, Instant startTime) throws PlanServiceException, IOException {
+    final var request = """
+       query{
+       span(where: {duration: {_is_null: true}, dataset_id: {_eq: %d}}) {
```

Comment:
Why do we specifically want spans with `null` durations here?

Reply:
Spans with the null durations are the unfinished activities (?)

Reply:
Oh! The method name confused me.